src/ucp/tag/rndv.c

```diff
@@ -363,11 +363,9 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_proto_progress_rndv_get_zcopy, (self),
                    ep, rndv_req, rndv_req->send.lane);
 
     /* rndv_req is the internal request to perform the get operation */
-    if ((rndv_req->send.state.dt.dt.contig[0].memh == UCT_MEM_HANDLE_NULL) &&
-        (ucp_ep_md_attr(ep, rndv_req->send.lane)->cap.flags & UCT_MD_FLAG_NEED_MEMH)) {
+    if (ucp_ep_md_attr(ep, rndv_req->send.lane)->cap.flags & UCT_MD_FLAG_NEED_MEMH) {
         /* TODO Not all UCTs need registration on the recv side */
-        UCS_PROFILE_REQUEST_EVENT(rndv_req->send.rndv_get.rreq, "rndv_recv_reg", 0);
-        status = ucp_request_send_buffer_reg(rndv_req, rndv_req->send.lane);
+        status = ucp_request_send_buffer_reg_lane(rndv_req, rndv_req->send.lane);
```

Comment:
it will try to re-reg memory on every iteration... may be set any flag to avoid multiple re-registration checks?

Reply:
yes! it checks md_map to avoid duplicate registration